Source/Core/DolphinQt/GameList/GameList.cpp

```diff
@@ -628,6 +640,60 @@ void GameList::OpenGCSaveFolder()
     ModalMessageBox::information(this, tr("Information"), tr("No save data found."));
 }
 
+void GameList::AddShortcutToDesktop()
+{
+  const auto game = GetSelectedGame();
+  std::string game_name = game->GetLongName();
+  std::string file_path = game->GetFilePath();
+  std::string file_name = game->GetFileName();
+  std::wstring dolphin_path = QCoreApplication::applicationFilePath().toStdWString();
+  LPCWSTR dolphin_path_convert = dolphin_path.c_str();
+
+  // Arguments
+  std::string args = "-e \"" + file_path + "\"";
+  std::wstring args_convert = std::wstring(args.begin(), args.end());
+
+  // Gets the desktop position with shorcut
+  WCHAR* desktop = 0;
+  SHGetKnownFolderPath(FOLDERID_Desktop, KF_FLAG_NO_ALIAS, NULL, &desktop);
+  CoTaskMemFree(desktop);
+  // Converts desktop path to string
+  std::wstring ws = std::wstring(desktop);
+  std::string desktop_path = std::string(ws.begin(), ws.end());
+  desktop_path = desktop_path + "\\" + game_name + ".lnk";
+  LPCSTR pathLink = desktop_path.c_str();
+
+  HRESULT hres;
+  IShellLink* psl;
```

Comment:
You may wish to use [ComPtr](https://docs.microsoft.com/en-us/cpp/cppcx/wrl/comptr-class?view=msvc-160) here, so that the release calls further down below aren't necessary.